hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -2831,28 +2830,27 @@ public List<RepeatedOmKeyInfo> listTrash(String volumeName,
   public List<SnapshotInfo> listSnapshot(
       String volumeName, String bucketName, String snapshotPrefix,
       String prevSnapshot, int maxListResult) throws IOException {
-    if (isAclEnabled) {
-      omMetadataReader.checkAcls(ResourceType.BUCKET, StoreType.OZONE,
-          ACLType.LIST, volumeName, bucketName, null);
-    }
-    boolean auditSuccess = true;
-    Map<String, String> auditMap = buildAuditMap(volumeName);
-    auditMap.put(OzoneConsts.BUCKET, bucketName);
+    metrics.incNumSnapshotLists();
+    Map<String, String> auditMap = new HashMap<>();
     try {
-      metrics.incNumSnapshotLists();
-      return metadataManager.listSnapshot(volumeName, bucketName,
-          snapshotPrefix, prevSnapshot, maxListResult);
+      if (isAclEnabled) {
+        omMetadataReader.checkAcls(ResourceType.BUCKET, StoreType.OZONE,
+            ACLType.LIST, volumeName, bucketName, null);
+      }
+      auditMap = buildAuditMap(volumeName);
+      auditMap.put(OzoneConsts.BUCKET, bucketName);
```

Comment:
If ACL check rejects the request, audit log will not have volume/bucket names.  Not broken by this change, but I think it can be fixed in the same patch.  Only ACL check should be moved inside the `try` block.

Reply:
Fixed it.